pkg/resourcehandlers/git/git_resource_handler.go

```diff
@@ -150,53 +157,90 @@ func (g *Git) Accept(uri string) bool {
 	rl, err := github.Parse(node.NodeSelector.Path)
 	if err != nil {
+		if _, ok := err.(resourcehandlers.ErrResourceNotFound); ok {
+			return []*api.Node{}, nil
+		}
 		return nil, err
 	}
-	repositoryPath := g.repositoryPathFromResourceLocator(rl)
-	if err := g.prepareGitRepository(ctx, repositoryPath, rl); err != nil {
+
+	//preparing repository
+	if err := g.prepareGitRepository(ctx, rl); err != nil {
 		return nil, err
 	}
 
-	nodesSelectorLocalPath := g.getNodeSelectorLocalPath(repositoryPath, rl)
-	fileInfo, err := g.fileReader.Stat(nodesSelectorLocalPath)
-	if err != nil {
-		if g.fileReader.IsNotExist(err) {
-			return nil, resourcehandlers.ErrResourceNotFound(node.NodeSelector.Path)
-		}
+	return github.BaseResolveNodeSelector(ctx, rl, g, g.cache, node, excludePaths, frontMatter, excludeFrontMatter, depth)
+}
+
+type treeExtractor struct {
+	gitRH *Git
+}
+
+func (tE *treeExtractor) ExtractTree(ctx context.Context, rl *github.ResourceLocator) ([]*github.ResourceLocator, error) {
+	//preparing repository
+	repositoryPath := tE.gitRH.repositoryPathFromResourceLocator(rl)
+	if err := tE.gitRH.prepareGitRepository(ctx, rl); err != nil {
 		return nil, err
 	}
-	if !fileInfo.IsDir() && filepath.Ext(fileInfo.Name()) == ".yaml" {
-		return nil, fmt.Errorf("nodeSelector path is neither directory or module")
-	}
-	_node := &api.Node{
-		Nodes: []*api.Node{},
-	}
-	nb := &nodeBuilder{
-		rootNodePath:    nodesSelectorLocalPath,
-		someMap:         make(map[string]*api.Node),
-		resourceLocator: rl,
-	}
-	filepath.Walk(nodesSelectorLocalPath, nb.build)
-	for path, n := range nb.someMap {
-		parentPath := filepath.Dir(path)
-		parent, exists := nb.someMap[parentPath]
-		if !exists { // If a parent does not exist, this is the root.
-			_node = n
-		} else {
-			n.SetParent(parent)
-			n.Parent().Nodes = append(n.Parent().Nodes, n)
+	nodesSelectorLocalPath := tE.gitRH.getNodeSelectorLocalPath(repositoryPath, rl)
+	//var files []string
+	//var files2 []string
+	result := make([]*github.ResourceLocator, 0)
+	root := strings.Split(nodesSelectorLocalPath, rl.SHAAlias)[0] + rl.SHAAlias
+	err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if path == root {
+			return nil
+		}
+		relativePath := strings.TrimPrefix(path, root+"/")
+		//dont include files from the .git folder, because they are not part of the repository
+		if relativePath == ".git" || strings.HasPrefix(relativePath, ".git/") {
+			return nil
+		}
+		//files = append(files, relativePath)
+		resourceType := github.Blob
+		if info.IsDir() {
+			resourceType = github.Tree
 		}
+		result = append(result, &github.ResourceLocator{
+			Scheme:   rl.Scheme,
+			Host:     rl.Host,
+			Owner:    rl.Owner,
+			Repo:     rl.Repo,
+			SHAAlias: rl.SHAAlias,
+			Type:     resourceType,
+			Path:     relativePath,
+		})
+		return nil
+	})
+	if err != nil {
+		return nil, err
 	}
+	/*
```

Comment:
It is not a good practice to leave a commented code.
If you do not need this, you can just remove it